src/kit.ts

```diff
@@ -806,35 +806,61 @@ export async function scanForKits(opt?: KitScanOptions) {
     location: vscode.ProgressLocation.Notification,
     title: localize('scanning.for.kits', 'Scanning for kits'),
   };
+
   return vscode.window.withProgress(prog, async pr => {
     const isWin32 = process.platform === 'win32';
+
     pr.report({message: localize('scanning.for.cmake.kits', 'Scanning for CMake kits...')});
-    let scanPaths: string[] = [];
+    let scan_paths = [] as string[];
+
     // Search directories on `PATH` for compiler binaries
-    const pathvar = process.env['PATH']!;
-    if (pathvar) {
+    if (process.env.hasOwnProperty('PATH')) {
       const sep = isWin32 ? ';' : ':';
-      scanPaths = scanPaths.concat(pathvar.split(sep));
+      scan_paths = scan_paths.concat((process.env.PATH as string).split(sep));
     }
 
     // Search them all in parallel
-    let prs = [] as Promise<Kit[]>[];
+    let kit_promises = [] as Promise<Kit[]>[];
     if (isWin32 && kit_options.minGWSearchDirs) {
-      scanPaths = scanPaths.concat(convertMingwDirsToSearchPaths(kit_options.minGWSearchDirs));
+      scan_paths = scan_paths.concat(convertMingwDirsToSearchPaths(kit_options.minGWSearchDirs));
     }
-    const compiler_kits = scanPaths.map(path_el => scanDirForCompilerKits(path_el, pr));
-    prs = prs.concat(compiler_kits);
+
+    const compiler_kits = scan_paths.map(path_el => scanDirForCompilerKits(path_el, pr));
+    kit_promises = kit_promises.concat(compiler_kits);
+
     if (isWin32) {
-      const vs_kits = scanForVSKits(pr);
+      // Prepare clang-cl search paths
+      const clang_cl_paths = new Set<string>();
 
-      const clang_cl_path = ['C:\\Program Files (x86)\\LLVM\\bin', 'C:\\Program Files\\LLVM\\bin', ...scanPaths];
-      const clang_cl_kits = await scanForClangCLKits(clang_cl_path);
-      prs.push(vs_kits);
-      prs = prs.concat(clang_cl_kits);
+      // LLVM_ROOT environment variable location
+      if (process.env.hasOwnProperty('LLVM_ROOT')) {
+        const llvm_root = path.normalize(process.env.LLVM_ROOT as string + "\\bin");
+        clang_cl_paths.add(llvm_root);
+      }
+      // Default installation locations
+      clang_cl_paths.add('C:\\Program Files (x86)\\LLVM\\bin');
+      clang_cl_paths.add('C:\\Program Files\\LLVM\\bin');
+      // PATH environment variable locations
+      scan_paths.forEach(path_ => clang_cl_paths.add(path_));
+      // LLVM bundled in VS locations
+      const vs_installs = await vsInstallations();
+      const bundled_clang_cl_paths = vs_installs.map(vs_install => {
+        return vs_install.installationPath + "\\VC\\Tools\\Llvm\\bin";
+      });
+      bundled_clang_cl_paths.forEach(path_ => {clang_cl_paths.add(path_);});
+
+      // Scan for kits
+      const vs_kits = scanForVSKits(pr);
```

Comment:
What does changing the scanning order do?  `scanForVSKits` used to be first.  Is there a reason the order is changing?

Reply:
I just grouped the scanFor... functions together as in that way it is easier for me to follow the process flow. It doesn't change the extension's behavior.